docs/src/pages/theming.mdx

```diff
@@ -0,0 +1,430 @@
+Both `UploadButton` and `UploadDropzone` accept `appearance` prop. It accepts an
+object with keys that correspond to elements of component. You can pass
+following types of values:
+
+- `string` - it will be applied as a class to the element
+- `CSSProperties` object - it will be applied as an inline style to the element
+- `callback` - it will be called with current state of the component and should
+  return one of the above types
+
+The interfaces of `appearance` prop for `UploadButton` and `UploadDropzone` are
+following
+
+```ts
+type UploadButtonProps = {
+  /* rest of props */
+  appearance?: {
+    container?: StyleField;
+    button?: StyleField;
+    allowedContent?: StyleField;
+  };
+};
+
+type UploadDropzoneProps = {
+  /* rest of props */
+  appearance?: {
+    container?: StyleField;
+    uploadIcon?: StyleField;
+    label?: StyleField;
+    allowedContent?: StyleField;
+    button?: StyleField;
+  };
+};
+```
+
+The types of callback arguments are following:
+
+```ts
+type ButtonCallbackArguments = {
+  ready: boolean;
+  isUploading: boolean;
+  uploadProgress: number;
+  fileTypes: string[];
+};
+
+type DropzoneCallbackArguments = {
+  ready: boolean;
+  isUploading: boolean;
+  uploadProgress: number;
+  fileTypes: string[];
+  isDragActive: boolean;
+};
+```
+
+## How to theme
+
+### With TailwindCSS
+
+#### Configuring TailwindCSS
+
+To leverage best developer experience, we strongly recommend extending your
+TailwindCSS config using our utility function `withUt`
+
+<Callout type="warning">
+  Note: `withUt` is optional but if you do not use it, you will not be able to
+  use new classes and variants it adds. Also, in case you are not using it, make
+  sure to import default stylesheet into your app.
+</Callout>
+
+```ts filename="tailwind.config.ts"
+import { withUt } from "@uploadthing/react/tw";
+
+/** @type {import('tailwindcss').Config} */
+export default withUt({
+  // your config goes here
+});
+```
+
+```js filename="tailwind.config.js"
+// @ts-check
+const { withUt } = require("@uploadthing/react/tw");
+
+/** @type {import('tailwindcss').Config} */
+module.exports = withUt({
+  // your config goes here
+});
+```
+
+<Callout type="warning">
+  Note: Once you extended your config, it is recommended to remove default
+  stylesheet import
+</Callout>
+
+`withUt` does two major things:
+
+- Adds new entry to `content` property so that it includes UploadThing's
+  components. It allows to avoid having duplicated styles in your bundle.
+- Registers plugin and adds nice-to-have classes to improve styling experience.
+
+#### `className` prop
+
+`className` prop accepts any classes so you can pass there anything you like.
+When it comes to TailwindCSS, to make styling easier, `withUt` adds few utility
+classes and variants for you to leverage.
+
+Tailwind variants that allow applying styles to internal elements and their
+arbitrary variant alternatives:
+
+| Custom variant       |                        Arbitrary variant |
```

Comment:
Why should users care about this table?

Reply:
I thought it might be useful for particular users to know what arbitrary variants can be used instead of custom classes and variants. Though thinking about it now, I guess components anatomy should be enough to understand how to target children of component's container